model/src/main/proto/oppia_logger.proto

```diff
@@ -368,6 +375,30 @@ message EventLog {
     string profile_id = 2;
   }
 
+  // Structure of a feature flag context.
+  message FeatureFlagContext {
+    // The unique UUID of the user.
+    string unique_user_uuid = 1;
+
+    // The id of the current session.
+    string session_id = 2;
```

Comment:
Do we need this? Generally session_id is only used for user studies.

Reply:
Session ID in this case will help cross-reference logs made in the same session with the feature flag states of that session. So if a crash happens, we can easily check the feature flags log with the same session id and user id and then use this information to analyze the crash.

Reply:
Was that in the original project spec? I don't remember writing that requirement, actually. My main concern is that our session ID generation doesn't guarantee the level of uniqueness that we'd want in order to log it for all of our users.

Reply:
The session id will not be used in isolation. It will be used in combination with the UUID which should be sufficiently unique enough. The UUID will be used to identify the user and the session id will identify the session and its feature flag values.

Reply:
I think that makes sense, however I think that ``sessionId`` is the wrong thing to be using here. That ID is designed to only be used for granular tracking for user studies, and it also gets regenerated after a certain amount of inactivity (whereas feature flags only get logged on startup because they can't change across app instances).

It seems like what you actually want here is a new value which is essentially a "app_open_instance_id" that's probably time-based and generated exactly once per app open.

Also in order to correlate this correctly against all other events, you'd need to log it as a non-sensitive value as part of the properties of all events, not just the feature flag list event.

Reply:
To address this, I have created a new appSessionId variable in the LoggingIdentifierController and this will be used to track the app session from app launch to app destroyed. This will also be logged with other logs in the future.

Reply:
App session ID is a better fit. Could you please add a TODO for session ID on #5341 to utilize the snowflake-esque algorithm as defined in the TDD for both IDs.

Separately, I think you also mentioned a desire to use a prefix for app session ID--I think that's a good idea to incorporate now since we may want to change this prefix when we move over the snowflake-esque ID scheme (since it will look very different than UUID).

Reply:
I added the todos on the ApplicationLifecycleObserver and the ApplicationLifecycleObserverTest tests.